internal/edgeapi/backend/api.go

```diff
@@ -0,0 +1,68 @@
+package backend
+
+import (
+	"context"
+
+	"github.com/project-flotta/flotta-operator/models"
+)
+
+const (
+	// Registered describes edge device that is fully authorized to communicate with the control plane
+	Registered = RegistrationStatus("registered")
+	// Unregistered describes edge device that is not authorized to communicate with the control plane and will be
+	// instructed to execute unregistration logic
+	Unregistered = RegistrationStatus("unregistered")
+	// Unknown signals that the status of the device can't be established (for example due to data retrieval errors)
+	Unknown = RegistrationStatus("unknown")
+
+	// RetryContextKey is a context key for a bool value describing whether the call is a retry call
+	RetryContextKey = "retry"
+)
+
+type RegistrationStatus string
+
+// EdgeDeviceBackend represents API provided by data storage service to support edge device lifecycle.
+type EdgeDeviceBackend interface {
+	// GetRegistrationStatus responds with status of a device registration: {registered, unregistered, unknown}
+	GetRegistrationStatus(ctx context.Context, name, namespace string) (RegistrationStatus, error)
+
+	// GetConfiguration provides complete Edge Device configuration that should be applied to the device
+	GetConfiguration(ctx context.Context, name, namespace string) (*models.DeviceConfigurationMessage, error)
+
+	// Enrol records device willingness to be connected to the cluster.
+	Enrol(ctx context.Context, name, namespace string, enrolmentInfo *models.EnrolmentInfo) (bool, error)
+
+	// GetTargetNamespace returns the namespace the device should belong to. This method may return NotApproved error.
+	GetTargetNamespace(ctx context.Context, name, namespace string, matchesCertificate bool) (string, error)
```

Comment:
I am a bit confused by this method: is that performing the registration? As [here](https://github.com/project-flotta/flotta-operator/blob/1730ecc9c18db898283f54368f4908c0c7f5f292/internal/edgeapi/yggdrasil/yggdrasil.go#L220) it looks like it is the case

Reply:
Ok, it's the `FinalizeRegistration` that is actually performing the registration then

Reply:
That is correct - that method is responsible for making sure that device contains all the information to be seen as registered.
In case of existing k8s implementation it just adds finalizers and adds status information (not really required for the device to work properly).

Reply:
So why not simply call is `Register` as you did with `Enrol`?

Reply:
Done. Thank you for the suggestion!